### DISCUSSION TITLE
AI searchmachine is not working anymore

### DISCUSSION THREAD

**Discussion by canedje** [immich-app/immich#7568] [Q&A]
### The bug

Since last update The new AI search engine did work very well.
This morning I did see Immich regenerate all images (automaticly??)
A lot off jobs did run.
After that the AI search is not working anymore
Not coming up with images belonging to AI search.
Most of the time only one image. Mostly also not the right search topic

### The OS that Immich Server is running on

Geekom IT13 as a docker under Unraid

### Version of Immich Server

v1.97.0

### Version of Immich Mobile App

not used

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
```
docker run
  -d
  --name='immich'
  --net='ewhnet'
  --cpuset-cpus='2,4,6,8,10,3,5,7,9'
  -e TZ="Europe/Berlin"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="Tower2"
  -e HOST_CONTAINERNAME="immich"
  -e 'DB_HOSTNAME'='192.168.2.240'
  -e 'DB_USERNAME'='xxxxxxx'
  -e 'DB_PASSWORD'='xxxxxxxx'
  -e 'DB_DATABASE_NAME'='immich'
  -e 'REDIS_HOSTNAME'='192.168.2.240'
  -e 'DB_PORT'='5433'
  -e 'REDIS_PORT'='6379'
  -e 'REDIS_PASSWORD'='redis'
  -e 'MACHINE_LEARNING_WORKERS'='1'
  -e 'MACHINE_LEARNING_WORKER_TIMEOUT'='120'
  -e 'PUID'='99'
  -e 'PGID'='100'
  -e 'UMASK'='022'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8080]'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/imagegenius/templates/main/unraid/img/immich.png'
  -p '8089:8080/tcp'
  -v '/mnt/remotes/NAS_NetwerkSchijf/Media/immich/photos/':'/photos':'rw,slave'
  -v '/mnt/remotes/NAS_NetwerkSchijf/Media/immich/import/':'/import':'ro,slave'
  -v '/mnt/user/appdata/immich':'/config':'rw' 'ghcr.io/imagegenius/immich:latest'
38c0fe7d2c238851fca4cb2cae6f6b8c75d321f8372d8d06485bf54594f4b54f

The command finished successfully!
```
```

### Your .env content

```Shell
I don't know what this is.
This is the LOG:
text  error  warn  system  array  login  
```

[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/server-info/config, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/server-info/statistics, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/server-info/media-types, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/server-info/admin-onboarding, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] SharedLinkController {/api/shared-link}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/me, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/:id, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/:id, PATCH} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/:id, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/:id/assets, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/shared-link/:id/assets, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] SystemConfigController {/api/system-config}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/system-config, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/system-config/defaults, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/system-config, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/system-config/storage-template-options, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/system-config/map/style.json, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] TagController {/api/tag}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id, PATCH} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id/assets, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id/assets, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/tag/:id/assets, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] TrashController {/api/trash}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/trash/empty, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/trash/restore, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/trash/restore/assets, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] UserController {/api/user}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/info/:id, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/me, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/profile-image, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/:id, DELETE} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/:id/restore, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/profile-image, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/user/profile-image/:id, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RoutesResolver] PersonController {/api/person}:
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id/reassign, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id, PUT} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id/statistics, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id/thumbnail, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id/assets, GET} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [RouterExplorer] Mapped {/api/person/:id/merge, POST} route
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [NestApplication] Nest application successfully started
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [ImmichServer] Immich Server is listening on http://[::1]:8080 [v1.97.0] [PRODUCTION] 
[Nest] 189  - 03/01/2024, 8:16:12 PM     LOG [CommunicationRepository] Websocket Connect:    g8QKZvi21oeAutrgAAAB
[ig-init] done.
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [CommunicationRepository] Initialized websocket server
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [SystemConfigService] LogLevel=log (set via system config)
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [MetadataRepository] Initializing metadata repository
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [MetadataService] Initialized local reverse geocoder
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [NestFactory] Starting Nest application...
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] BullModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] ConfigHostModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] ScheduleModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] ConfigModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] BullModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] BullModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] TypeOrmCoreModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] InfraModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] DomainModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [InstanceLoader] MicroservicesModule dependencies initialized
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [NestApplication] Nest application successfully started
[Nest] 219  - 03/01/2024, 8:16:16 PM     LOG [ImmichMicroservice] Immich Microservices is listening on http://[::1]:3002 [v1.97.0] [PRODUCTION] 
[03/01/24 20:16:16] INFO     Starting gunicorn 21.2.0                           
[03/01/24 20:16:16] INFO     Listening at: http://0.0.0.0:3003 (214)            
[03/01/24 20:16:16] INFO     Using worker: uvicorn.workers.UvicornWorker        
[03/01/24 20:16:16] INFO     Booting worker with pid: 267                       
[03/01/24 20:16:19] INFO     Started server process [267]                       
[03/01/24 20:16:19] INFO     Waiting for application startup.                   
[03/01/24 20:16:19] INFO     Created in-memory cache with unloading after 300s  
                             of inactivity.                                     
[03/01/24 20:16:19] INFO     Initialized request thread pool with 20 threads.   
[03/01/24 20:16:19] INFO     Application startup complete.                      
[Nest] 189  - 03/01/2024, 8:19:07 PM     LOG [CommunicationRepository] Websocket Disconnect: g8QKZvi21oeAutrgAAAB
[Nest] 189  - 03/01/2024, 8:19:07 PM     LOG [CommunicationRepository] Websocket Connect:    ERujtW4omGfixsViAAAD

```
```

### Reproduction steps

```bash
1.just try to search
2.
3.
...
```

### Additional information

It did work fine at the last release
![image](https://github.com/immich-app/immich/assets/23150947/c4c3539b-a4ca-4f22-b171-5790c1fa343e)

**Comment by mertalev** [comment]
This happens as a side effect of running thumbnail generation. This side effect will be removed in the next release, but it won't help you in this case since it already ran. If you know how to query Postgres, you can run `VACUUM FULL smart_search;` to fix it. If not, re-running smart search on all assets through the job dashboard will also do the same thing (but take more time).

**Comment by canedje** [comment]
Thanks for the quick answer, so I have to wait for the next release.
I now did already force to regenerate all images (41.500!!!) so it will be busy for some time.

**Comment by canedje** [comment]
After total rebuild of all images AI search still not working
After VACUUM FULL smart_search; it is working again

**Reply by mertalev** [reply]
Glad to hear it! You might also want to run `VACUUM FULL asset_faces;` to clean up that table too. It'll help with facial recognition

**Comment by canedje** [comment]
thanks again
